src/main/java/com/hazelcast/remotecontroller/HzCluster.java

```diff
@@ -34,8 +34,12 @@ public HzCluster(String version, String xmlConfig) {
         } else {
             this.config = new XmlConfigBuilder().build();
         }
-        //configure cluster id as cluster name
-        config.setClusterName(id);
+        if (!keepClusterName) {
```

Comment:
Shouldn't it be `if (keepClusterName)`, considering that `id` is set to what's in the config if this condition is `true`?

Reply:
exactly 👍  fixed it